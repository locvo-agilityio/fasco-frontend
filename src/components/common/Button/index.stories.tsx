import { StoryObj, Meta } from '@storybook/react';

// Components
import Button from '.';

const meta = {
  title: 'Components/Common/Button',
  component: Button,
  parameters: {
    layout: 'centered',
  },

  render: (args) => (
    <div className="w-32">
      <Button {...args}> Click</Button>
    </div>
  ),
  tags: ['autodocs'],
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof Button>;

export const Default: Story = {};

export const Primary: Story = {
  args: {
    color: 'primary',
    variant: 'ghost',
    size: 'lg',
  },
};

export const Secondary: Story = {
  args: {
    color: 'primary',
    variant: 'ghost',
    size: 'lg',
    className: 'text-foreground-500 border-primary-300',
  },
};

export const Ternary: Story = {
  args: {
    color: 'primary',
    variant: 'ghost',
    size: 'md',
    className: 'text-foreground-500 py-2',
  },
};

export const Outlined: Story = {
  args: {
    color: 'primary',
    size: 'md',
    variant: 'ghost',
    className:
      'border-none data-[hover=true]:!bg-primary-300 data-[hover=true]:!text-primary-foreground',
  },
};
