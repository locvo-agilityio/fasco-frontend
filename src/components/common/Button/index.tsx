'use client';

import { ComponentProps, memo } from 'react';
import { extendVariants, Button as ButtonNextUI } from '@heroui/react';

// Utils
import { cn } from '@/utils';

export const ButtonBase = extendVariants(ButtonNextUI, {
  variants: {
    variant: {
      bordered: 'border-1',
      ghost: 'bg-transparent data-[hover=true]:!border-none',
    },

    color: {
      default: 'bg-primary-300 text-foreground-100 disabled:opacity-50',
      primary:
        'text-secondary-400 border border-secondary-100 disabled:opacity-50',
    },

    size: {
      md: 'w-full py-5 text-base font-poppins',
      lg: 'w-full py-5 text-base font-poppins font-semibold',
    },
    radius: {
      none: 0,
      md: 'rounded-base',
      lg: 'rounded-regular',
      full: 'rounded-full',
    },
  },

  defaultVariants: {
    radius: 'md',
    variant: 'bordered',
    color: 'default',
    size: 'md',
  },
});

const Button = ({ className, ...props }: ComponentProps<typeof ButtonBase>) => (
  <ButtonBase
    className={cn('data-[pressed=true]:scale-[1] font-poppins', className)}
    {...props}
  />
);

export default memo(Button);
