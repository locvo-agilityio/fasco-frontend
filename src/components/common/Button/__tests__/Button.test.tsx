import { render } from '@testing-library/react';

import Button from '..';

describe('Button', () => {
  it('should render snapshot', () => {
    const { container } = render(<Button>Default Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it('renders with primary color', () => {
    const { getByRole } = render(<Button color="primary">Primary</Button>);

    const buttonPrimary = getByRole('button');

    expect(buttonPrimary).toHaveClass(
      'text-secondary-400 border border-secondary-100 disabled:opacity-50',
    );
  });
});
