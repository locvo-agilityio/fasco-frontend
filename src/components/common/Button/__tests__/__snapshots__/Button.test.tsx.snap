// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render snapshot 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none border-default data-[hover=true]:opacity-hover border-1 bg-primary-300 text-foreground-100 disabled:opacity-50 w-full py-5 text-base rounded-base data-[pressed=true]:scale-[1] font-poppins"
    type="button"
  >
    Default Button
  </button>
</div>
`;
